### PR TITLE
Feature: Create user profile page

### PR DESCRIPTION
**Pull request by weihesdlegend**
## Description
Create user profile page to display saved travel plans in reverse chronological order.

## Solution
* Serve the page from backend using Go HTML templates
* Create delete personal travel plan endpoint

## Testing
Integration tests with Heroku testing app

## Final Checks
- [x] Have you removed commented code ?
- [x] Have you used gofmt to format your code ?
- [x] You have added unit tests

### CODE REVIEW

**Review by rwangsc18 — COMMENTED** [review]
LGTM, have a few questions for my own understanding purpose.
I will check the display issue of profile_page.html in mobile screen size.

File: iowrappers/users.go [hunk 1]
```diff
@@ -148,6 +148,31 @@ func (redisClient *RedisClient) SaveUserPlan(context context.Context, userView u
...
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
+	go redisClient.findUserPlan(context, redisKey, &planView, &wg)
+
+	wg.Wait()
+
+	userSavedPlansRedisKey := strings.Join([]string{UserSavedTravelPlansPrefix, "user", userView.ID, "plans"}, ":")
+	travelPlanRedisKey := strings.Join([]string{TravelPlanRedisCacheKeyPrefix, planView.OriginalPlanID}, ":")
+	if res, originalPlanIdRemovalErr := redisClient.client.SRem(context, userSavedPlansRedisKey, travelPlanRedisKey).Result(); originalPlanIdRemovalErr != nil && originalPlanIdRemovalErr != redis.Nil {
```
[the remark this answers is not in the document]
Comment (weihesdlegend) [reply iowrappers/users.go:168]: Yes we could, but it is idiomatic in Go to localize the variables initialized in if statements.

File: iowrappers/users.go [hunk 2]
```diff
@@ -148,6 +148,31 @@ func (redisClient *RedisClient) SaveUserPlan(context context.Context, userView u
...
+	if findUserErr != nil {
+		return findUserErr
+	}
+
+	redisKey := strings.Join([]string{UserSavedTravelPlanPrefix, "user", userView.ID, "plan", planView.ID}, ":")
+	Logger.Debugf("plan to be deleted: %s", redisKey)
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
+	go redisClient.findUserPlan(context, redisKey, &planView, &wg)
```
[the remark this answers is not in the document]
Comment (weihesdlegend) [reply iowrappers/users.go:162]: `FindUserPlans` is used externally to find all the plans for a user, while `findUserPlan` is an internal method that fetches just one plan. We could use `FetchSingleRecord` to refactor it though.

File: iowrappers/users.go [hunk 3]
```diff
@@ -148,6 +148,31 @@ func (redisClient *RedisClient) SaveUserPlan(context context.Context, userView u
...
+	wg.Wait()
+
+	userSavedPlansRedisKey := strings.Join([]string{UserSavedTravelPlansPrefix, "user", userView.ID, "plans"}, ":")
+	travelPlanRedisKey := strings.Join([]string{TravelPlanRedisCacheKeyPrefix, planView.OriginalPlanID}, ":")
+	if res, originalPlanIdRemovalErr := redisClient.client.SRem(context, userSavedPlansRedisKey, travelPlanRedisKey).Result(); originalPlanIdRemovalErr != nil && originalPlanIdRemovalErr != redis.Nil {
+		Logger.Infof("result from removing original key from hash set is %d", res)
+		return originalPlanIdRemovalErr
+	}
+
+	return redisClient.RemoveKeys(context, []string{redisKey})
```
[the remark this answers is not in the document]
Comment (weihesdlegend) [reply iowrappers/users.go:173]: Not really, the `SRem` removes a single value in a HashSet without removing the Redis key, while the `RemoveKeys` removes the whole data structures behind the keys completely.

[the remark this answers is not in the document]
Comment (rwangsc18) [reply iowrappers/users.go:173]: Got it, thanks for the explanation.

**Review by rwangsc18 — APPROVED** [review]
Approved

**Pull request merged**